plugins/MauticCrmBundle/Command/PushActivitiesToPipedriveCommand.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace MauticPlugin\MauticCrmBundle\Command;
+
+use Mautic\CoreBundle\Helper\ProgressBarHelper;
+use Mautic\PluginBundle\Entity\IntegrationEntity;
+use MauticPlugin\MauticCrmBundle\Integration\Pipedrive\Export\ActivitiesPipedriveExport;
+use MauticPlugin\MauticCrmBundle\Integration\Pipedrive\Export\LeadExport;
+use MauticPlugin\MauticCrmBundle\Integration\PipedriveIntegration;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class PushActivitiesToPipedriveCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('mautic:integration:pipedrive:pushactivities')
+            ->setDescription('Pushes the data from Mautic to Pipedrive')
+            ->addOption(
+                'time-interval',
+                't',
+                InputOption::VALUE_OPTIONAL,
+                'Send time interval to check updates, it should be a correct php formatted time interval in the past eg:(15 minutes)',
+                '15 minutes'
+            );
+
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $integrationHelper = $this->getContainer()->get('mautic.helper.integration');
+        /** @var PipedriveIntegration $integrationObject */
+        $integrationObject = $integrationHelper->getIntegrationObject(PipedriveIntegration::INTEGRATION_NAME);
+        $em                = $this->getContainer()->get('doctrine')->getManager();
+
+        if (!$integrationObject->getIntegrationSettings()->getIsPublished()) {
+            $output->writeln('Pipedrive integration id disabled.');
+
+            return;
+        }
+
+        $output->writeln('<comment>Push activities</comment>');
+
+        /** @var ActivitiesPipedriveExport $activitiesExport */
+        $activitiesExport = $this->getContainer()->get('mautic_integration.pipedrive.export.activities');
+        $activitiesExport->setIntegration($integrationObject);
+
+        $pushed = 0;
+        $start  = 0;
+        $limit  = 50;
+
+        // now() - time-interval
+        $timeInterval     = str_replace('--', '-', '-'.$input->getOption('time-interval'));
+        $lastPossibleSync = (new \DateTime())->setTimestamp(strtotime($timeInterval));
+
+        $progress = ProgressBarHelper::init($output, $limit);
```

Comment:
The second param should be max count (the count of 100%). 50 is a wild guess of how many integration entities there are. You should count the amount of integration entities that will be processed instead.

Reply:
The issue is we don't know how many items will be processed.   
We store lastActivitySync timestamp to internal serialize column.
This is reason why we have to run all integration entity in batch, then just skipped already updated   https://github.com/mautic/mautic/pull/7233/files#diff-6bdf0596bca5f7bff88ba72d06c7683dR78

I know It's not the best solution, but I didn't better idea how to figure out.
Maybe we can use regexp on intetrnal column and this should work too.